### PR TITLE
DOCSP-30337: Add Manage Sync Session - Kotlin SDK page

### PR DESCRIPTION
**Pull request by cbullinger**
## Pull Request Info

Add new Manage Sync Session page for Kotlin (progress listener added in https://github.com/realm/realm-kotlin/pull/1118)

> NOTE: Kotlin SDK doesn't currently support progress listener for Flexible Sync; only supported for PBS

### Jira

- https://jira.mongodb.org/browse/DOCSP-30337

### Staged Changes

- [Manage Sync Session](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-30337-manage-sync-sessions/sdk/kotlin/sync/manage-sync-session/) - new page
- [Partition-Based Sync](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-30337-manage-sync-sessions/sdk/kotlin/sync/partition-based-sync/#check-upload---download-progress-for-a-sync-session) - updated page

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/kotlin/sync/manage-sync-session.txt [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+#. :ref:`Configure Flexible Sync on the Atlas App Services backend <enable-flexible-sync>`.
+#. :ref:`Authenticate a user <kotlin-authenticate>` in your client app.
+#. :ref:`Open the synced realm <kotlin-open-a-synced-realm>`.
+
+.. _kotlin-sync-wait-for-changes:
+
+Wait for Changes to Upload and Download
+---------------------------------------
+
+To asynchronously wait for all changes to upload to Atlas from your synced realm,
```
Comment (cmelchior) [inline source/sdk/kotlin/sync/manage-sync-session.txt:41]: I would probably also try to add a sentence or two about timeouts, I.e. if you don't want to let the user hang forever you can do `downloadAllRemoteChanges(1.minute)` after which an exception is thrown (data is still being downloaded in the background but you might want to change the app behavior if things are slow.

File: source/sdk/kotlin/sync/partition-based-sync.txt [hunk 2]
```diff
@@ -59,7 +58,34 @@ To adjust specific configuration settings, use the options provided by
 
 .. literalinclude:: /examples/generated/kotlin/SyncTest.snippet.configure-a-synced-realm.kt
     :language: kotlin
-    :copyable: false
+
+Check Upload & Download Progress for a Sync Session
+---------------------------------------------------
+
+To monitor Sync upload and download progress, call
```
Comment (cmelchior) [inline source/sdk/kotlin/sync/partition-based-sync.txt:65]: Should we mention that this isn't supported for Flexible Sync yet?

**Review by cmelchior — COMMENTED** [review]
The code and content look fine to me. I added a few suggestions for improvements.

File: source/sdk/kotlin/sync/partition-based-sync.txt (hunk 2 above)
Comment (cbullinger) [reply source/sdk/kotlin/sync/partition-based-sync.txt:65]: Yep. I'll add a note

File: source/sdk/kotlin/sync/manage-sync-session.txt (hunk 1 above)
Comment (cbullinger) [reply source/sdk/kotlin/sync/manage-sync-session.txt:41]: Updated

File: source/sdk/kotlin/sync/manage-sync-session.txt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+This method returns ``true`` when all changes have been uploaded. 
+
+To asynchronously wait for all changes on Atlas to download from the Device Sync 
+server to your synced realm, call 
+`downloadAllServerChanges <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb.sync/-sync-session/download-all-server-changes.html>`__.
+This method returns ``true`` when all changes have been downloaded.
+
+You can also include an optional ``timeout`` parameter to either method to 
+determine the maximum amount of time before an exception is thrown. Note that 
+the upload or download continues in the background even after returning ``false``.
```
Comment (cmelchior) [inline source/sdk/kotlin/sync/manage-sync-session.txt:53]: Sorry, I was a bit quick. Double-checking our implementation and docs. We do not throw an exception for a timeout, but do return `false`

**Review by cmelchior — APPROVED** [review]

File: source/sdk/kotlin/sync/manage-sync-session.txt (hunk 3 above)
Comment (cbullinger) [reply source/sdk/kotlin/sync/manage-sync-session.txt]: Thanks. Corrected

**Review by dacharyc — APPROVED** [review]
One of the API reference links is 403-ing for me, but otherwise this LGTM overall.

File: source/sdk/kotlin/sync/partition-based-sync.txt [hunk 4]
```diff
@@ -59,7 +58,40 @@ To adjust specific configuration settings, use the options provided by
...
+`progressAsFlow() <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb.sync/-sync-session/progress-as-flow.html>`__ 
+
+This method returns a Flow of 
+`Progress <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb.sync/-progress/index.html>`__ 
+events. ``Progress`` provides the total 
+number of transferrable bytes and the remaining bytes to be transferred.
+
+``syncSession.progressAsFlow()`` takes two arguments:
+
+- A `Direction <{+kotlin-sync-prefix+}/io.realm.kotlin.mongodb.sync/-direction/index.html>`__
```
[the remark this answers is not in the document]
Comment (cbullinger) [reply source/sdk/kotlin/sync/partition-based-sync.txt:81]: Fixed

File: source/sdk/kotlin/sync/partition-based-sync.txt [hunk 5]
```diff
@@ -59,7 +58,40 @@ To adjust specific configuration settings, use the options provided by
...
+---------------------------------------------------
+
+.. note:: 
+
+  The ``progressAsFlow()`` listener in the Kotlin SDK is only currently 
+  available for realms using Partition-Based Sync. The Kotlin SDK does not 
+  yet support progress listeners for Flexible Sync.
+
+To monitor Sync upload and download progress, call 
+`progressAsFlow() <{+kotlin-sync-prefix+}io.realm.kotlin.mongodb.sync/-sync-session/progress-as-flow.html>`__
```
[the remark this answers is not in the document]
Comment (cbullinger) [reply source/sdk/kotlin/sync/partition-based-sync.txt:72]: Updated

**Pull request merged**